### DISCUSSION TITLE
Hardware Acceleration possible?

### DISCUSSION THREAD

**Discussion by KleinRiese** [ptr727/PlexCleaner#118] [General]
Hi, I'm using PlexCleaner (Docker build, shared /dev/dri) and it seems that there is no hardware acceleration.
Is my system too old (Celeron N3050)? Or is hardware acceleration not possible?

**Comment by ptr727** [comment]
I've not tested or accounted for any hardware video encoding using GPU's.
I know ffmpeg and handbrake supports hardware acceleration, but I also know that passing GPU's to docker is non trivial.

See:
https://github.com/NVIDIA/nvidia-docker
https://handbrake.fr/docs/en/latest/technical/video-nvenc.html
https://trac.ffmpeg.org/wiki/HWAccelIntro

**Comment by KleinRiese** [comment]
Okay - but maybe you would give it a try? :)
At least with Linux, acceleration is no big deal for intel / amd. Basicly you just have to share /dev/dri and observe the access rights.

Currently you use ffmpeg and handbrake, handbrake is just a failsafe here?

For ffmpeg, encoder would be "h264_qsv", that's all - a little bit more configurable {VideoCodec} would be enough.
Unfortunately I don't know handbrake :(

**Comment by ptr727** [comment]
Unfortunately I do not have any quicksync or nvidia hardware to try with, but you are welcome to try by modifying the commandline options in the code.
Handbrake is sometimes a fallback if ffmpeg encoding fails, but handbrake is always used for deinterlacing.

**Reply by KleinRiese** [reply]
What, I have to do it by myself? Just kidding! Thank you for your work!

I'm not very experienced, but I'll give it a try. Currently I'm facing problems because vaapi isn't available.
Seems to be a good work for the colder days...

**Reply by ptr727** [reply]
Can you try Beta 3, I added the ability to set custom options.

**Reply by KleinRiese** [reply]
Wow, you made it? Thank you! I didn't managed to get it working.
I'll give it a try next week.

**Comment by ptr727** [comment]
Try out v3 Beta.
